Framework/include/QualityControl/ActivityHelpers.h

```diff
@@ -22,64 +22,59 @@
 #include <algorithm>
 #include <map>
 #include <string>
+#include <ranges>
 #include <boost/property_tree/ptree_fwd.hpp>
 
 namespace o2::quality_control::core::activity_helpers
 {
 
-namespace implementation
+template <typename Range>
+concept RangeOfActivities = requires(Range range)
 {
+  requires std::same_as<std::ranges::range_value_t<Range>, Activity>;
+};
 
-template <typename Iter, typename Accessor = const Activity& (*)(const Activity&)>
-void commonActivityFields(Iter begin, Iter end, Accessor accessor, Activity& result);
-
+namespace implementation
+{
+Activity commonActivityFields(const RangeOfActivities auto& activities);
 }
 
 /// Produces the most constrained Activity which will match all the provided
-template <typename Iter, typename Accessor = const Activity& (*)(const Activity&)>
-Activity strictestMatchingActivity(
-  Iter begin, Iter end, Accessor accessor = [](const Activity& a) -> const Activity& { return a; })
```

Comment:
why was accessor removed?

Reply:
The accessor was added in the first place because we did not have a way to easily pass some kind of iterator of all the activities if they are members bigger structures. `std::ranges` allow for that, as it is demostrated in the test, so we can remove this and expect a `RangeOfActivities` instead.